pddlgym/backend/state.py

```diff
@@ -0,0 +1,274 @@
+                    raise ValueError("Predicate {} is not defined in the state.".format(goal.name))
+                domain.check_types(goal.types)
+        self.special_effects = special_effects
+        
+    def __eq__(self, other):
+        """
+        Checks if two states are equal.
+
+        Args:
+            other (State): The state to compare to.
+
+        Returns:
+            bool: True if the states are equal, False otherwise.
+        """
+        return self.domain == other.domain and self.objects == other.objects \
+            and self.predicates == other.predicates and self.goal == other.goal \
+            and self.special_effects == other.special_effects
+
+    def check_predicate(self, predicate, value=True):
+        """
+        Checks if a predicate is in the state with the correct value.
+
+        Args:
+            predicate (Predicate): The predicate to check.
+            value (bool): The value of the predicate to check for.
+
+        Returns:
+            bool: True if the predicate is in the state with the correct 
+            value, False otherwise.
+        """
+        return predicate in self.predicates and self.predicates[predicate] == value
+        
+    def update_predicate(self, predicate, value):
+        """
+        Updates a predicate in the state with a new value.
+
+        Args:
+            predicate (Predicate): The predicate to update.
+            value (bool): The new value of the predicate.
+        """
+        if predicate not in self.predicates:
+            raise ValueError("Predicate {} is not in the state.".format(predicate))
+        self.predicates[predicate] = value
+    
+    def update_special_effect(self, special_effect, obj, args):
+        """
+        Updates a special effect in the state.
+        If the special effect is not in the state, it is added.
+        The special effect is then updated as an active update.
+
+        Args:
+            special_effect (SpecialEffect): The special effect to update.
+            obj (Object): The object to update the special effect for.
+            args (Dictionary[Object, Object]): The arguments for the special
+                effect.
+        """
+        new_special_effect = special_effect.copy(obj, args)
+        if new_special_effect not in self.special_effects:
+            self.special_effects.append(new_special_effect)
+        current = self.special_effects[self.special_effects.index(new_special_effect)]
+        current.update(self, active=True)
+        if current.completed:
+                self.special_effects.remove(current)
+
+    def check_goal(self):
+        """
+        Checks if the state satisfies the goal.
+
+        Returns:
+            bool: True if the state satisfies the goal, False otherwise.
+        """
+        satisfied = False
+        for goal_set in self.goal:
+            for goal in goal_set:
+                if not self.check_predicate(goal):
+                    break
```

Comment:
no need for breaks, only returns. rewrite this function where the last return statement is `return False` and then figure out where you would put a `return True` inside without breaking.